### ISSUE TITLE
lsl_viewer needs 3x bandwidth

### ISSUE THREAD

**Issue by behinger** [sccn/labstreaminglayer#122]
Edit: I had the epiphany to just monitor my network bandwidth (iftop). I saw that lsl_recorder takes ~2.6mbit, so similar to what I measured (why is it smaller, who knows)

But lsl_viewer needs 11mbit! much more than I anticipated. So what's going on with lsl_viewer - I will move this to the respective git repo.

Sorry for bothering! Leason learned is to not assume a program is written in a way you'd think it is written :D

-----
Old message
Hi!
I noticed in my custom chunked LSL stream (via wifi, but >20mbit capacity easily reached) of 129channels (~30ms) sampled at 1000kHz Float32 only reaches the nominal sampling rate, when maximally 2 inlets are active (e.g. lsl_viewer and lsl_recorder). As soon as I add e.g.  `lsl_status`, the effective sampling rate drops by 30-40% and the resulting XDF has dropped samples (my EEG amplifier sends a sample-counter, so I really know they are dropped).

For my lsl receiving config, I used the wifi-recommended settings

My calculation on throughput (via iperf, I reached ~30mbit/s) are: 128x1000x32 are 4 mbit/s - thus I should be able to easily accomodate 7 concurrent streams (I'm assuming LSL is not as smart as to pull the package and then locally distribut it again ;)).

Any ideas? I need at least one stream for visualizing EEG, and one for recording. But it would be nice to have a e.g. data-quality measurement in parallel.

I tried to simplify and reduce my (unholy, I admit I know little c++) code to the minimum - maybe I missed something.
 
```C++
lsl::stream_info data_info("eegosport", "EEG", nChannels, this->samplingRate, lsl::cf_float32, "eegoSports_" + amp->getType());
	
// while loop
        usleep(30000); // faster polling is not better
        buf = amplifierStream->getData();
  	double now = lsl::local_clock();
        ampleCount = buf.size() / channelCount;
       
        std::vector<std::vector<float>> send_buffer(sampleCount, std::vector<float>(nChannels));
        for (int s = 0; s < sampleCount; s++) {
	        for (int c = 0; c < nChannels; c++) {
                      send_buffer[s][c] = microvoltUnit * buf.getSample(c, s);
	}}

// Code to fill the buffer // 
		data_outlet.push_chunk(send_buffer, now);
```
